File: 2021/day01/nim/day01.nim
import std/strutils
import std/sequtils

proc count_increases(items : seq[int]) : int = 
  for i in 1 .. items.high:
    if items[i] > items[i - 1]:
      inc result

proc get_sums(items: seq[int]) : seq[int] =
    for i in countup(2,items.len-1):
        result.add(items[i-2]+items[i-1]+items[i])

let input: seq[int] = "../input.txt".lines.toSeq.map(parseInt)
echo "Part1: ",input.count_increases()
echo "Part2: ",input.get_sums().count_increases()

File: 2019/day01/day01.rb
def fuel(f)
    sum=0
    while f>0 do
        f=f/3-2
        sum+=f if f>0
    end
    sum
end

input=File.read('input.txt').lines 
part1=input.map{ |l| (l.to_i / 3) - 2}.sum 
puts "Part1=#{part1}"

part2=input.map{ |l| fuel(l.to_i)}.sum
puts "Part2=#{part2}"